AudioContextInterruptedState/explainer.md

```diff
@@ -0,0 +1,212 @@
+
+## The `"interrupted"` state
+
+This explainer proposes adding the `"interrupted"` state to the `AudioContextState` enum, as shown below:
+
+```js
+enum AudioContextState {
+    "suspended",
+    "running",
+    "closed",
+    "interrupted"
+};
+```
+
+Whenever an `AudioContext` transitions to either `"suspended"` or `"interrupted"`, audio playback would halt. The main difference between `"suspended"` and `"interrupted"` is that an `AudioContext` can only move to the `"suspended"` state if the [user triggered](https://webaudio.github.io/web-audio-api/#dom-audiocontext-suspended-by-user-slot) the state change; while the UA can transition the context to the `"interrupted"` state if there is a need for that.
+
+With the addition of the `"interrupted"` state, the following state transitions would also be introduced:
+- `"running"` -> `"interrupted"`;
+  - Would happen whenever the UA needs to interrupt audio playback.
+- `"suspended"` -> `"interrupted"`;
+  - Shouldn't happen automatically. This transition should happen only if there is an ongoing interruption and [`AudioContext.resume()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-resume) is called.
+- `"interrupted"` -> `"running"`;
+  - By the time that the cause of the interruption ceases to exist, the UA can transition to `"running"` if audio playback is allowed to resume automatically.
+- `"interrupted"` -> `"suspended"`; and
+  - By the time that the cause of the interruption ceases to exist, the UA can transition to `"suspended"` if audio playback is **not** allowed to resume automatically or if [`AudioContext.suspend()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-suspend) has been called during the interruption.
+- `"interrupted"` -> `"closed"`.
+  - The `AudioContext`'s state should move immediately to `"closed"` when `AudioContext.close()` is called.
+
+The state transition from `"suspended"` to `"interrupted"` should not happen automatically, due to privacy concerns. Doing this transition automatically every time an interruption occurs might unnecessarily expose too much information to web pages - e.g., when a phone call comes in. Given this, what should happen if an interruption happens while the `AudioContext` is in the `"suspended"` state? In this case, an internal boolean flag, let's say `"is_interrupted"`, should be set and the AudioContext should remain in the `"suspended"` state, even though it is interrupted behind the curtains. However, calling [`AudioContext.resume()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-resume) should trigger the state change from `"suspended"` to `"interrupted"` if the interruption is still active. Therefore, while in the `"suspended"` state with an ongoing interruption:

```

Comment:
Would it be simpler to remove the rhetorical question and simply state our strategy here?

`To prevent fingerprinting in this scenario, an internal boolean flag, let's say ....`